lib/charms/opensearch/v0/opensearch_peer_clusters.py

```diff
@@ -394,45 +393,54 @@ def demote_to_failover_orchestrator(self) -> None:
             Scope.APP, "deployment-description", deployment_desc.to_dict()
         )
 
-    def validate_roles(self, nodes: List[Node], on_new_unit: bool = False) -> None:
-        """Validate full-cluster wide the quorum for CM/voting_only nodes on services start."""
+    def has_recommended_cm_count(self, nodes: List[Node]) -> bool:
+        """Validate cluster-wide count for CM-eligible nodes is at least 3"""
         deployment_desc = self.deployment_desc()
-        if not set(deployment_desc.config.roles) & {"cluster_manager", "voting_only"}:
-            # the user is not adding any cm nor voting_only roles to the nodes
-            return
-
-        if deployment_desc.start == StartMode.WITH_GENERATED_ROLES:
-            # the roles are automatically generated, we trust the correctness
+        if (
+            not set(deployment_desc.config.roles) & {"cluster_manager", "voting_only"}
+            and deployment_desc.start != StartMode.WITH_GENERATED_ROLES
+        ):
+            # only CM-eligible nodes run the validations
+            return True
+
+        is_large_deployment = self.is_provider() or self.is_consumer()
+        cms = sum(1 for node in nodes if node.is_cm_eligible())
+        if not is_large_deployment or (is_large_deployment and cms >= 3):
+            # validation is only needed in large deployments
+            # we can start with any number of cms but set a blocked status
+            # if CMS < 3 when a CM-eligible application is scaled down or removed
+            return True
+
+        logger.info("Less than 3 CM-eligible units in this cluster")
+        return False
+
+    def validate_recommended_cm_unit_count(
+        self, nodes: Optional[List[Node]] = None, only_validate_if_blocked: bool = False
```

Comment:
It is not very clear what is meant by `only_validate_if_blocked`, could you please rename the argument?